crates/lib/src/cli.rs

```diff
@@ -859,7 +914,11 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
             .unwrap_or_default();
         if staged_unchanged {
             println!("Staged update present, not changed.");
-
+            if opts.queue_soft_reboot {
+                if has_soft_reboot_capability(host.status.staged.as_ref()) {
+                    soft_reboot_staged(sysroot)?;
+                }
```

Comment:
The silent fallthrough here makes me I wonder if we should actually have `--soft-reboot=required|auto` or so where `required` fails if the reboot wouldn't be soft? Implementing this would change from a `bool` to an `Option<enum SoftRebootRequest { Auto, Required }>`

That'd allow higher level tooling which might operate under a "minimize downtime" constraint to impose that requirement.

Reply:
Latest push attempts this, It seems to work I need still to comb over all the changes to make sure I am not missing something but should good for today's demo.
https://asciinema.org/a/PvBJXqlNiqRdf3suYLukLYsSu